### DISCUSSION TITLE
AutoGPT with LocalAI

### DISCUSSION THREAD

**Discussion by Chalo2023** [mudler/LocalAI#300] [Ideas]
Hello everyone!

LocalAI is a great idea, it would be better if it had an installer to avoid installation errors....what is the procedure to use AutoGPT with LocaAI?
Greetings
Marco

**Comment by watcher60** [comment]
I think this does what you want ...however I have failed to get it to generate anything. Please note my terrible hack (in the closed issues of the below repo) to extend the timeout value for the API calls. It might be my setup but interested if you get anywhere with it. If using linux you will need to manually install golang and make before running the script,
https://github.com/aorumbayev/autogpt4all

**Comment by Chalo2023** [comment]
Of course, I have also seen other projects but I have not asked to make 
them work, let me know if you can:
- https://github.com/gravelBridge/AutoGPT-BingAI
- https://github.com/Significant-Gravitas/Auto-GPT-Plugins
- https://github.com/cognosisai/platform
Greetings,

El 2023-05-19 04:37, watcher60 escribió:

> I think this does what you want ...however I have failed to get it to 
> generate anything. Please note my terrible hack (in the closed issues 
> of the below repo) to extend the timeout value for the API calls. It 
> might be my setup but interested if you get anywhere with it. If using 
> linux you will need to manually install goland and make before running 
> the script,
> https://github.com/aorumbayev/autogpt4all
> 
> --
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>


Links:
------
[1] 
https://github.com/go-skynet/LocalAI/discussions/300#discussioncomment-5946668
[2]

**Reply by watcher60** [reply]
I'm sorry I'm not following you - you asked about the process to get LocalAI installed with Auto-GPT, the project I linked to is exactly :  _"AutoGPT4All is a simple bash script that sets up and configures [AutoGPT](https://github.com/Significant-Gravitas/Auto-GPT.git) running with the [GPT4All](https://github.com/aorumbayev/autogpt4all#) model on the [LocalAI](https://github.com/go-skynet/LocalAI) server. This setup allows you to run queries against an open-source licensed model without any limits, completely free and offline.". I'm completely unclear why you responded as you did.  

On a side note moving the setup to a dedicated Ubuntu PC from being a wsl ubuntu instance seems to have drastically improved my experience, though malformed json in AutoGPT is still an issue (as it is for everyone else running AutoGPT it seems),

**Comment by Chalo2023** [comment]
Hello, thanks for your answer, apparently I got confused by the google 
translator......I try to use autoGPT without API KEY or find an 
AutoGPT4ALL project....I send you the links of the projects related to 
autoGPT without API KEY that I found in case it helps... I don't have 
programming knowledge, so I can't install the projects correctly... will 
you have the installation process for LocalAI + autoGPT + GPT4all?
Greetings

El 2023-05-19 10:45, watcher60 escribió:

> I'm sorry I'm not following you - you asked about the process to get 
> LocalAI installed with Auto-GPT, the project I linked to is exactly : 
> _"AutoGPT4All is a simple bash script that sets up and configures 
> AutoGPT [1] running with the GPT4All [2] model on the LocalAI [3] 
> server. This setup allows you to run queries against an open-source 
> licensed model without any limits, completely free and offline.". I'm 
> completely unclear why you responded as you did.
> 
> On a side note moving the setup to a dedicated Ubuntu PC from being a 
> wsl ubuntu instance seems to have drastically improved my experience, 
> though malformed json in AutoGPT is still an issue (as it is for 
> everyone else running AutoGPT it seems),
> 
> --
> Reply to this email directly, view it on GitHub [4], or unsubscribe 
> [5].
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>


Links:
------
[1] https://github.com/Significant-Gravitas/Auto-GPT.git
[2] https://github.com/aorumbayev/autogpt4all#
[3] https://github.com/go-skynet/LocalAI
[4] 
https://github.com/go-skynet/LocalAI/discussions/300#discussioncomment-5950108
[5]

**Comment by Chalo2023** [comment]
Try to install the project:
https://github.com/cryptocake/Auto-GPT-LLaMA
I got the following error:

El 2023-05-19 10:45, watcher60 escribió:

> I'm sorry I'm not following you - you asked about the process to get 
> LocalAI installed with Auto-GPT, the project I linked to is exactly : 
> _"AutoGPT4All is a simple bash script that sets up and configures 
> AutoGPT [1] running with the GPT4All [2] model on the LocalAI [3] 
> server. This setup allows you to run queries against an open-source 
> licensed model without any limits, completely free and offline.". I'm 
> completely unclear why you responded as you did.
> 
> On a side note moving the setup to a dedicated Ubuntu PC from being a 
> wsl ubuntu instance seems to have drastically improved my experience, 
> though malformed json in AutoGPT is still an issue (as it is for 
> everyone else running AutoGPT it seems),
> 
> --
> Reply to this email directly, view it on GitHub [4], or unsubscribe 
> [5].
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>


Links:
------
[1] https://github.com/Significant-Gravitas/Auto-GPT.git
[2] https://github.com/aorumbayev/autogpt4all#
[3] https://github.com/go-skynet/LocalAI
[4] 
https://github.com/go-skynet/LocalAI/discussions/300#discussioncomment-5950108
[5]

**Comment by Aisuko** [comment]
Hi, guys. Here we already supported AutoGPT. https://localai.io/integrations/autogpt4all/index.html
